src/components/map/EventLayer.js

```diff
@@ -169,13 +169,18 @@ class EventLayer extends Layer {
                     ];
 
                     if (displayEl) {
-                        let value = dataValue.value;
-
-                        if (displayEl.optionSet) {
-                            value = displayEl.optionSet[value];
+                        const { valueType, optionSet, name } = displayEl;
+                        let { value } = dataValue;
+
+                        if (valueType === 'COORDINATE' && value) {
+                            value = JSON.parse(value)
```

Comment:
This will throw an error if the server doesn't return valid JSON, we should move it to another function and wrap it in a try/catch

Reply:
Wrapped in a try/catch